applicationset/generators/scm_provider.go

```diff
@@ -127,6 +127,7 @@ func (g *SCMProviderGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 			"url":              repo.URL,
 			"branch":           repo.Branch,
 			"sha":              repo.SHA,
+			"short_sha":        repo.SHA[:8],
```

Comment:
Should there be a length check, just in case the API returns an empty (or short) string?

Reply:
Same as above